Content.Server/Xenoarchaeology/Artifact/XenoArtifactSystem.cs

```diff
@@ -20,6 +25,31 @@ private void OnArtifactMapInit(Entity<XenoArtifactComponent> ent, ref MapInitEve
     {
         if (ent.Comp.IsGenerationRequired)
             GenerateArtifactStructure(ent);
+
+        // DeltaV - start of research statistics admin logs
+        foreach (var node in GetAllNodes(ent))
+        {
+            var effectStatusString = node.Comp.LockedEffectTipHidden ? "Hidden" : node.Comp.LockedEffectTipVague ? "Vague" : "Specific";
+
+            var triggerStr = new StringBuilder();
+            var predecessors = GetPredecessorNodes((ent, ent), node);
+            triggerStr.Append(predecessors.Count + 1);
+            triggerStr.Append(" triggers: ");
+            triggerStr.Append(node.Comp.TriggerTip ?? "Unknown");
+            foreach (var predecessor in predecessors)
+            {
+                triggerStr.Append(",");
+                triggerStr.Append(predecessor.Comp.TriggerTip ?? "Unknown");
+            }
+
+            _adminLogger.Add(
+                LogType.ArtifactNode,
+                LogImpact.Low,
+                // note: effect type is already logged by ToPrettyString(node)
+                $"{ToPrettyString(ent.Owner)} spawned with node {ToPrettyString(node)} with depth {node.Comp.Depth}; effect status {effectStatusString}; {triggerStr.ToString()}"
+            );
```

Comment:
This one seems like it would get really noisy. It would spit out an admin log for each node when an artifact spawns, right?

Reply:
That's correct. There are 10-16 nodes in each artifact. It'll shoot off 10-16 logs back to back whenever an artifact spawns.

I guess I'm not totally sure how many artifacts get generated in a round. I imagined 6 would be an average number, but maybe it's more like 10 with how many salvage wrecks there are? Surely not more than 20 or 30? (30 artifacts would be only 400 logs per round)

Reply:
I think a lot more may get generated, though not necessarily utilized. There are multiple that spawn on red rock at round start and possibly on salv wrecks as well.

Reply:
We could add some intelligent filtering on "only log the artifacts that are actually used", but I think that might be premature.

My goal with this particular "what triggers does a node have" log is so we can reconstruct the full trees. Seeing what nodes researchers *didn't* unlock can be just as useful as seeing which ones they did -- I'd like to try to analyze the actual/perceived difficulty of trigger types by using these logs.